### DISCUSSION TITLE
Allow Sorting Lists by No. of Repositories in each List

### DISCUSSION THREAD

**Discussion by sohang3112** [community/community#40765] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

It would be nice to be able to sort Lists (which shows on the top of the Stars page) by the number of repositories in each List.
In fact, IMO this should probably be the default behaviour - show the Lists on top of the Stars page in Descending order (based on No. of Repositories in each List).

**Comment by radar-blip** [comment]
Good idea! 👍🏻

**Reply by wizdommusa** [reply]
Reason[]()

**Comment by ossanna16** [comment]
Hi there @sohang3112 and welcome to our community! Thank you for the suggestion!

**Reply by Abhisheknavaale** [reply]
def test_calendar():
    print("RUNNING TEST_CALENDAR()")
    test_event1 = {"start": {"date": "2022-01-01"}, "end": {"date": "2022-01-07"}, "summary":"test event 1"}
    test_event2 = {"start": {"date": "2022-02-01"}, "end": {"date": "2022-02-07"}, "summary":"test event 2"}
    events = [test_event1, test_event2]

    return events
test_calendar()
    context = {"results": results}
    return render(request, 'demo.html', context

<!DOCTYPE html>
<html>
  <head> </head>
  <body>
    <h1>Demo</h1>
    <ul>
      {% for result in results %}
      <li>{{result.start.date}}{% if result.end.date %}-{% endif%}{{result.end.date}}: {{result.summary}}</li>
      {% endfor %}
    </ul>
  </body>
</html>
{
"type": "service_account",
"project_id": "example",
"private_key_id": "BLAHBALH",
"private_key": "-----BEGIN PRIVATE KEY-----\nSUUUUUUPER------LONG-------STRING\n-----END PRIVATE KEY-----\n",
"client_email": "USERNAME@EXAMPLE.iam.gserviceaccount.com",
"client_id": "1234567890987654321",
"auth_uri": "https://accounts.google.com/o/oauth2/auth",
"token_uri": "https://oauth2.googleapis.com/token",
"auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
"client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/username%40example.iam.gserviceaccount.com"
}
# this hidden folder contains your local, virtual environment
.env

# this hidden file contains sensitive keys and environmental config vars
# if you've named your primary project folder something other than
# 'example' please adjust the following line as needed 
example/.env

# this token contains your sensitive google service account info
google-credentials.json

from decouple import config
from google.oauth2 import service_account
import googleapiclient.discovery
import datetime

CAL_ID = config('CAL_ID')
SCOPES = ['https://www.googleapis.com/auth/calendar']
SERVICE_ACCOUNT_FILE = './google-credentials.json'


def test_calendar():
    print("RUNNING TEST_CALENDAR()")

    credentials = service_account.Credentials.from_service_account_file(SERVICE_ACCOUNT_FILE, scopes=SCOPES)
    service = googleapiclient.discovery.build('calendar', 'v3', credentials=credentials)

    # CREATE A NEW EVENT
    new_event = {
    'summary': "Ben Hammond Tech's Super Awesome Event",
    'location': 'Denver, CO USA',
    'description': 'https://benhammond.tech',
    'start': {
        'date': f"{datetime.date.today()}",
        'timeZone': 'America/New_York',
    },
    'end': {
        'date': f"{datetime.date.today() + datetime.timedelta(days=3)}",
        'timeZone': 'America/New_York',
    },
    }
    service.events().insert(calendarId=CAL_ID, body=new_event).execute()
    print('Event created')

 # GET ALL EXISTING EVENTS
    events_result = service.events().list(calendarId=CAL_ID, maxResults=2500).execute()
    events = events_result.get('items', [])

    # LOG THEM ALL OUT IN DEV TOOLS CONSOLE
    for e in events:

        print(e)

    #uncomment the following lines to delete each existing item in the calendar
    #event_id = e['id']
        # service.events().delete(calendarId=CAL_ID, eventId=event_id).execute()


    return events

**Reply by Abhisheknavaale** [reply]
def test_calendar():
    print("RUNNING TEST_CALENDAR()")
    test_event1 = {"start": {"date": "2022-01-01"}, "end": {"date": "2022-01-07"}, "summary":"test event 1"}
    test_event2 = {"start": {"date": "2022-02-01"}, "end": {"date": "2022-02-07"}, "summary":"test event 2"}
    events = [test_event1, test_event2]

    return events
test_calendar()
    context = {"results": results}
    return render(request, 'demo.html', context

<!DOCTYPE html>
<html>
  <head> </head>
  <body>
    <h1>Demo</h1>
    <ul>
      {% for result in results %}
      <li>{{result.start.date}}{% if result.end.date %}-{% endif%}{{result.end.date}}: {{result.summary}}</li>
      {% endfor %}
    </ul>
  </body>
</html>
{
"type": "service_account",
"project_id": "example",
"private_key_id": "BLAHBALH",
"private_key": "-----BEGIN PRIVATE KEY-----\nSUUUUUUPER------LONG-------STRING\n-----END PRIVATE KEY-----\n",
"client_email": "USERNAME@EXAMPLE.iam.gserviceaccount.com",
"client_id": "1234567890987654321",
"auth_uri": "https://accounts.google.com/o/oauth2/auth",
"token_uri": "https://oauth2.googleapis.com/token",
"auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
"client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/username%40example.iam.gserviceaccount.com"
}
# this hidden folder contains your local, virtual environment
.env

# this hidden file contains sensitive keys and environmental config vars
# if you've named your primary project folder something other than
# 'example' please adjust the following line as needed 
example/.env

# this token contains your sensitive google service account info
google-credentials.json

from decouple import config
from google.oauth2 import service_account
import googleapiclient.discovery
import datetime

CAL_ID = config('CAL_ID')
SCOPES = ['https://www.googleapis.com/auth/calendar']
SERVICE_ACCOUNT_FILE = './google-credentials.json'


def test_calendar():
    print("RUNNING TEST_CALENDAR()")

    credentials = service_account.Credentials.from_service_account_file(SERVICE_ACCOUNT_FILE, scopes=SCOPES)
    service = googleapiclient.discovery.build('calendar', 'v3', credentials=credentials)

    # CREATE A NEW EVENT
    new_event = {
    'summary': "Ben Hammond Tech's Super Awesome Event",
    'location': 'Denver, CO USA',
    'description': 'https://benhammond.tech',
    'start': {
        'date': f"{datetime.date.today()}",
        'timeZone': 'America/New_York',
    },
    'end': {
        'date': f"{datetime.date.today() + datetime.timedelta(days=3)}",
        'timeZone': 'America/New_York',
    },
    }
    service.events().insert(calendarId=CAL_ID, body=new_event).execute()
    print('Event created')

 # GET ALL EXISTING EVENTS
    events_result = service.events().list(calendarId=CAL_ID, maxResults=2500).execute()
    events = events_result.get('items', [])

    # LOG THEM ALL OUT IN DEV TOOLS CONSOLE
    for e in events:

        print(e)

    #uncomment the following lines to delete each existing item in the calendar
    #event_id = e['id']
        # service.events().delete(calendarId=CAL_ID, eventId=event_id).execute()


    return events

**Reply by Nooknick7899** [reply]
events_result = service.events().list(calendarId=CAL_ID, maxResults=2500).execute()
events = events_result.get('items', [])

# LOG THEM ALL OUT IN DEV TOOLS CONSOLE
for e in events:

    print(e)

#uncomment the following lines to delete each existing item in the calendar
#event_id = e['id']
    # service.events().delete(calendarId=CAL_ID, eventId=event_id).execute()


return events

**Reply by Nooknick7899** [reply]
> def test_calendar ( ): พิมพ์ ( "RUNNING TEST_CALENDAR ( )" ) test_event1 = { "เริ่มต้น": { "date": "2022-01-01" }, "สิ้นสุด": { "date": "2022-01-07" }, "สรุป":" เหตุการณ์ทดสอบ <TAG1 test_event2 = { "เริ่มต้น": { "วันที่": "2022-02-01" }, "สิ้นสุด": { "วันที่": "2022-02-07" }, "สรุป":" เหตุการณ์ทดสอบ <TAG1 เหตุการณ์ = [ test_event1, test_event2 ]
> 
> ```
> return events
> ```
> 
> test_calendar ( ) บริบท = { "ผลลัพธ์": ผลลัพธ์ } ส่งคืนเรนเดอร์ ( re ⁇ uest, 'demo.html', บริบท
> 
> # Demo
> {% สําหรับผลลัพธ์ที่ได้% }
> * { { result.start.date } } {% ถ้า result.end.date% } - {% endif% } { result.end.date { }
> {% end สําหรับ% }
> 
> { "type": "service_account", "project_id": "ตัวอย่าง", "private_key_id": "BLAHBALH", "private_key": "----- BEGIN KEY ส่วนตัว ----- \ nSUUUUUPER ------- ยาว ------- STRING \ n ------ สิ้นสุดคีย์ส่วนตัว ----- \ n", "client_email": "USERNAME@E ⁇ AMPLE.iam.gserviceaccount.com", "client_id": "1234567890987654321", "auth_uri": "https://accounts.google.com/o/oauth2/auth", "token_uri": "https://oauth2.googleapis.com/token", "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs", "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/username%40example.iam.gserviceaccount.com" } # โฟลเดอร์ที่ซ่อนอยู่นี้มีโลคัลของคุณ,สภาพแวดล้อมเสมือน .env
> # ไฟล์ที่ซ่อนอยู่นี้มีคีย์ที่ละเอียดอ่อนและ vars การกําหนดค่าด้านสิ่งแวดล้อม
> # หากคุณตั้งชื่อโฟลเดอร์โครงการหลักของคุณอย่างอื่นที่ไม่ใช่
> # 'ตัวอย่าง' โปรดปรับบรรทัดต่อไปนี้ตามต้องการ
> ตัวอย่าง / .env
> 
> # โทเค็นนี้มีข้อมูลบัญชีบริการ google ที่ละเอียดอ่อนของคุณ
> google-credentials.json
> 
> จากการกําหนดค่าการนําเข้า decouple จาก google.oauth2 นําเข้า service_account นําเข้า googleapiclient.discovery นําเข้าวันที่
> 
> CAL_ID = config ( 'CAL_ID' ) SCOPES = [ 'https://www.googleapis.com/auth/calendar'] SERVICE_ACCOUNT_FILE = './google-credentials.json'
> 
> def test_calendar ( ): พิมพ์ ( "RUNNING TEST_CALENDAR ( )" )
> 
> ```
> credentials = service_account.Credentials.from_service_account_file(SERVICE_ACCOUNT_FILE, scopes=SCOPES)
> service = googleapiclient.discovery.build('calendar', 'v3', credentials=credentials)
> 
> # CREATE A NEW EVENT
> new_event = {
> 'summary': "Ben Hammond Tech's Super Awesome Event",
> 'location': 'Denver, CO USA',
> 'description': 'https://benhammond.tech',
> 'start': {
>     'date': f"{datetime.date.today()}",
>     'timeZone': 'America/New_York',
> },
> 'end': {
>     'date': f"{datetime.date.today() + datetime.timedelta(days=3)}",
>     'timeZone': 'America/New_York',
> },
> }
> service.events().insert(calendarId=CAL_ID, body=new_event).execute()
> print('Event created')
> ```
> 
> # รับกิจกรรมที่มีอยู่ทั้งหมด
> ```
> events_result = service.events().list(calendarId=CAL_ID, maxResults=2500).execute()
> events = events_result.get('items', [])
> 
> # LOG THEM ALL OUT IN DEV TOOLS CONSOLE
> for e in events:
> 
>     print(e)
> 
> #uncomment the following lines to delete each existing item in the calendar
> #event_id = e['id']
>     # service.events().delete(calendarId=CAL_ID, eventId=event_id).execute()
> 
> 
> return events
> ```

**Reply by SerhiyKhrus777777** [reply]
Copy

**Comment by Trinhan87** [comment]
> ### Select Topic Area
> 
> Product Feedback
> 
> ### Body
> 
> It would be nice to be able to sort Lists (which shows on the top of the Stars page) by the number of repositories in each List.
> In fact, IMO this should probably be the default behaviour - show the Lists on top of the Stars page in Descending order (based on No. of Repositories in each List).

**Comment by Nooknick7899** [comment]
events_result = service.events().list(calendarId=CAL_ID, maxResults=2500).execute()
events = events_result.get('items', [])

# LOG THEM ALL OUT IN DEV TOOLS CONSOLE
for e in events:

    print(e)

#uncomment the following lines to delete each existing item in the calendar
#event_id = e['id']
    # service.events().delete(calendarId=CAL_ID, eventId=event_id).execute()


return events

**Comment by sohang3112** [comment]
I haven't tried out the above solutions - but I meant that the GitHub team should add this to the site itself. Running seperate code just for this would be too much hassle!

**Comment by jfullstackdev** [comment]
what would be the benefit of it ? and it should be even the default behavior ?

**Comment by aioroy** [comment]
J

**Reply by sohang3112** [reply]
What does "J" mean??
